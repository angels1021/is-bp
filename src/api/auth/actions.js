/*
 * LoginActions
 *
 * redux actions to describe login/logout actions
 *
 * */
import { createAction } from 'redux-actions';
import { createFormAction } from 'redux-form-saga';
import {
  LOGIN_REQUEST,
  LOGOUT_REQUEST,
  AUTH_REQUEST,
  AUTH_ERROR,
  AUTH_SUCCESS,
  AUTH_SET
} from './constants';

/**
 * Tells the app we want to login a user
 * @param  {object} formData          The formData we're sending for login
 * @param  {string} formData.username The username of the user to login
 * @param  {string} formData.password The password of the user to login
 * @param  {string} formData.location The location to navigate to after login
 * @param  {string} formData.code     The card code of the user to login (optional)
 *
 * @return {object} action object - { type: LOGIN_REQUEST, payload: formData{} }
 */
export const login = createAction(LOGIN_REQUEST, (formData) => formData.toObject()); // (formData) => formData

/**
 * Tells the app we want to log out a user
 *
 * @param {string/number} userId - the user to logout;
 *
 * @return {object} action object - { type: LOGOUT_REQUEST, payload: userId }
 */
export const logout = createAction(LOGOUT_REQUEST); // (userId) => userId

/**
 * Sets the 'error' state to error received (both login and logout use the same reducer)
 * @see http://redux-form.com/6.5.0/docs/api/SubmissionError.md/
 * for SubmissionError
 *
 * @param {object} Error/SubmissionError The error received when trying to make a request.
 *
 * @return {object} action object - { type: AUTH_ERROR, payload: error }
 */
export const authError = createAction(AUTH_ERROR); // (SubmissionError) => SubmissionError
// export const authError = (error) => ({
//   type: AUTH_ERROR,
//   payload: error
// }); // (error) => error

/**
 * Sets the 'pending' state to requested action (both login and logout use the same reducer)
 *
 * @param {string} type The action type making the request
 *
 * @return {object} action object - { type: AUTH_REQUEST, payload: LOGIN_REQUEST }
 */
export const authRequest = createAction(AUTH_REQUEST); // (type) => type

/**
 * Sets the 'user' state to requested response (both login and logout use the same reducer)
 *
 * @param {object} [user={}] The new user object to set. on logout it's unnecessary
 * @param {string} user.id The user id
 * @param {string} user.uuid The user's uuid
 * @param {string} user.role The user's authorized role
 * @param {string} user.firstName The user's first name
 * @param {string} user.lastName The user's last name
 *
 * @return {object} action object - { type: AUTH_SUCCESS, payload: user{} }
 */
export const authSuccess = createAction(AUTH_SUCCESS, (user) => user || {});

/**
 * Sets the authentication state of the application
 *
 * @param  {boolean} isLoggedIn
 */
export const authSet = createAction(AUTH_SET, (isLoggedIn) => Boolean(isLoggedIn));

export const loginAction = createFormAction(login, [AUTH_SUCCESS, AUTH_ERROR]);
