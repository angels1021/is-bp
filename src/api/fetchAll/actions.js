/**
 * fetchAll api actions
 */
import { FETCH_ALL } from './constants';
import { createRequestId } from './utils';
/**
 * action creator for a fetchAll request.
 * expects an options object
 *
 * @param {string} requestId           - unique page/module id to track in global async reducer
 * @param {generator} fetchGenerator   - generator to call when ready to fetch resources
 * @param {string} [parentId]          - unique page/module id to wait for before starting calls
 * @param {function} [resolveSelector] - selector function that must resolve to true before starting calls
 */
export const fetchAllCreateSaga = ({ requestId, fetchGenerator, parentId, resolveSelector }) => ({
  type: FETCH_ALL,
  payload: {
    requestId: createRequestId(requestId),
    parentId: createRequestId(parentId),
    fetchGenerator,
    resolveSelector
  }
});

/**
 * actions creator for fetchAll events
 * should be listened to in individual module/page sagas
 *
 * @param moduleId
 */
export const fetchAll = (moduleId) => ({
  type: createRequestId(moduleId),
  payload: { id: moduleId }
});

