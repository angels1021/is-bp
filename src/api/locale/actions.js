/**
 * Locale api actions
 */
import {
  CHANGE_LOCALE_REQUEST,
  CHANGE_LOCALE_SUCCESS,
  CHANGE_LOCALE_FAIL,
  CHANGE_LOCALE_CANCEL
} from './constants';

/**
 * changeLocale - action creator
 *
 * @param {string} locale - two character language code, ie 'en'.
 * @param {object[]} request - request translations for the modules specified - passed to api/translations.
 * @property {string} request[].module - ie 'app'.
 * @property {string} request[].page ie  -'common'.
 * @param {string} moduleId - the caller module who's reducer should catch this actions
 */
export const changeLocale = (locale, request, moduleId) => ({
  type: CHANGE_LOCALE_REQUEST,
  payload: { locale, request, moduleId }
});

/**
 * changeLocaleSuccess - action creator
 *
 * @param {string} locale - two character language code, ie 'en'.
 * @param {string} moduleId - the caller module who's reducer should catch this actions
 */
export const changeLocaleSuccess = (locale, moduleId) => ({
  type: CHANGE_LOCALE_SUCCESS,
  payload: { locale, moduleId }
});

/**
 * changeLocaleFail - action creator
 *
 * @param {object} error - the response error
 * @param {string} moduleId - the caller module who's reducer should catch this actions
 */
export const changeLocaleFail = (error, moduleId) => ({
  type: CHANGE_LOCALE_FAIL,
  payload: { error, moduleId }
});

/**
 * changeLocaleCancel - action creator
 *
 * @param {string} moduleId - the caller module who's reducer should catch this actions
 */
export const changeLocaleCancel = (moduleId) => ({
  type: CHANGE_LOCALE_CANCEL,
  payload: { moduleId }
});

