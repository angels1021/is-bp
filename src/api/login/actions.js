/*
* LoginActions
*
* redux actions to describe login/register actions
*
* */
import {
  SET_AUTH,
  REQUEST_PENDING,
  LOGIN_REQUEST,
  LOGOUT,
  REQUEST_ERROR,
  REQUEST_SUCCESS,
  CLEAR_ERROR
} from './constants';

/**
* requestSuccess - set the user received from login
*
* @param {type} user Description
* @param  {string} user.id The user id
* @param  {string} user.uuid The user's uuid
* @param  {string} user.role The user's authorized role
* @param  {string} user.firstName The user's first name
* @param  {string} user.lastName The user's last name
*
* @returns {type} Description
*/
export const requestSuccess = (user) => ({
  type: REQUEST_SUCCESS,
  user
});

/**
 * Sets the authentication state of the application
 *
 * @param  {boolean} newAuthState True means a user is logged in, false means no user is logged in
 */
export const setAuth = (newAuthState) => ({
  type: SET_AUTH,
  newAuthState: Boolean(newAuthState)
});

/**
 * Sets the `currentlySending` state, which displays a loading indicator during requests
 *
 * @param  {boolean} busy True means we're currently sending a request, False means we're idle
 */
export const sendingRequest = (pending) => ({
  type: REQUEST_PENDING,
  pending: Boolean(pending)
});

/**
 * Tells the app we want to login a user
 * @param  {object} data          The data we're sending for login
 * @param  {string} data.username The username of the user to login
 * @param  {string} data.password The password of the user to login
 * @param  {string} data.location The location to navigate to after login
 * @param  {string} data.code     The card code of the user to login (optional)
 */
export const loginRequest = (data) => ({
  type: LOGIN_REQUEST,
  data
});

/**
 * Tells the app we want to log out a user
 */
export const logoutRequest = () => ({
  type: LOGOUT
});


/**
 * Sets the 'error' state to error received
 *
 * @param {object} error The error received when trying to make arequest
 */
export const requestError = (error) => ({
  type: REQUEST_ERROR,
  error
});

/**
 * Sets the 'error' state as empty
 */
export const clearError = () => ({
  type: CLEAR_ERROR
});
