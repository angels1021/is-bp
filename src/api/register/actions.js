/*
* LoginActions
*
* redux actions to describe login/register actions
*
* */

import {
  CHANGE_FORM,
  SENDING_REQUEST,
  REGISTER_REQUEST,
  REQUEST_ERROR,
  CLEAR_ERROR
} from './constants';

/**
 * Sets the form state
 *
 * @param  {object} newFormState          The new state of the form
 * @param  {string} newFormState.username The new text of the username input field of the form
 * @param  {string} newFormState.password The new text of the password input field of the form
 */
export const changeForm = (newFormState) => ({
  type: CHANGE_FORM,
  newFormState
});


/**
 * Sets the `currentlySending` state, which displays a loading indicator during requests
 *
 * @param  {boolean} busy True means we're sending a request, false means we're not
 */
export const sendingRequest = (busy) => ({
  type: SENDING_REQUEST,
  busy: Boolean(busy)
});

/**
 * Tells the app we want to register a user
 * @param  {object} data          The data we're sending for registration
 * @param  {string} data.username The username of the user to register
 * @param  {string} data.password The password of the user to register
 */
export const registerRequest = (data) => ({
  type: REGISTER_REQUEST,
  data
});


/**
 * Sets the 'error' state to error received
 *
 * @param {object} error The error received when trying to make arequest
 */
export const requestError = (error) => ({
  type: REQUEST_ERROR,
  error
});

/**
 * Sets the 'error' state as empty
 */
export const clearError = () => ({
  type: CLEAR_ERROR
});
