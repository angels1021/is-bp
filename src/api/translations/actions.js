/**
 * action creators for the translations api.
 */

import {
  MESSAGES_REQUEST,
  MESSAGES_SUCCESS,
  MESSAGES_FAIL
} from './constants';

/**
 * requestMessages - action creator
 * dispatch to request translations for each module
 *
 * @param {string} locale - two character language code, ie 'en'.
 * @param {object[]} request - request translations for the modules specified.
 * @property {string} request[].module - ie 'app'.
 * @property {string} request[].page ie  -'common'.
 */
export const requestMessages = (locale, request) => ({
  type: MESSAGES_REQUEST,
  payload: { locale, request }
});

/**
 * messageSuccess - action creator
 * dispatch to add received translations to state
 *
 * @param {string} locale - two character language code, ie 'en'.
 * @param {object[]} response - found translations for the modules requested.
 * @property {string} response[].module - ie 'app'.
 * @property {string} response[].page ie  -'common'.
 * @property {object} response[].messages ie  - the translation messages found
 */
export const messageSuccess = (locale, response) => ({
  type: MESSAGES_SUCCESS,
  payload: { locale, response }
});

/**
 * messagesError - action creator
 * dispatch if requestMessages was rejected
 *
 * @param {object} error - the response error
 */
export const messagesError = (error) => ({
  type: MESSAGES_FAIL,
  payload: { error }
});
