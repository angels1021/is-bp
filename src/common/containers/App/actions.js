/**
 * Global action creators
 *
 * here will setup action creators to be used by the global reducer
 * and common actions that apply to any component
 */
import { ASYNC_REQUEST, ASYNC_SUCCESS, ASYNC_FAIL } from './constants';

/**
 * action creator for ASYNC_REQUEST
 * called to start tracking an async call on the global object
 *
 * @param  {string} id - a unique request id to track
 */
export const asyncRequest = (id) => ({
  type: ASYNC_REQUEST,
  payload: { id }
});

/**
 * action creator for ASYNC_SUCCESS
 * called to complete tracking an async call and remove from global.loading list
 *
 * @param  {string} id - the unique request id used in asyncRequest
 */
export const asyncSuccess = (id) => ({
  type: ASYNC_SUCCESS,
  payload: { id }
});

/**
 * action creator for ASYNC_FAIL
 * called to complete tracking an async call and set an error in global.errors list
 *
 * @param {string} id    - the unique request id used in asyncRequest
 * @param {object} error - a new Error object to set as the request error
 */
export const asyncFail = (id, error) => ({
  type: ASYNC_FAIL,
  payload: { id, error }
});
