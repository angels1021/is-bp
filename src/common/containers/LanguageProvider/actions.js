/*
 *
 * LanguageProvider actions factory
 * since all pages share the same language structure
 * we can use a factory to create their actions
 *
 */

const LanguageActionsFactory = (
  CHANGE_LOCALE_REQUEST,
  CHANGE_LOCALE_SUCCESS,
  REQUEST_MESSAGES,
  MESSAGES_PENDING,
  MESSAGES_SUCCESS,
  MESSAGES_ERROR
) => ({
  /**
   * changeLocale action creator
   *
   * @param {string} locale - locale identifier to request
   * @param {object} request - locale modules request object
   * @property {string} module - name of the module to fetch
   * @property {string} page - name of the page messages to fetch
   */
  changeLocale: (locale, request) => ({
    type: CHANGE_LOCALE_REQUEST,
    locale,
    request
  }),


  /**
   * changeLocaleSuccess action creator
   *
   * @param {string} locale
   * @param {array} response
   * @property {object} responseItem
   * @property {string} responseItem.module
   * @property {string} responseItem.page
   * @property {object} responseItem.messages
   */
  changeLocaleSuccess: (locale, response) => ({
    type: CHANGE_LOCALE_SUCCESS,
    locale,
    response
  }),
  /**
   * requestMessages action creator
   *
   * @param {string} locale - locale identifier to request
   * @param {object} request - locale modules request object
   * @property {string} module - name of the module to fetch
   * @property {string} page - name of the page messages to fetch
   */
  requestMessages: (locale, request) => ({
    type: REQUEST_MESSAGES,
    locale,
    request
  }),

  /**
   * requestPending action creator
   *
   * set the messages state as pending
   */
  requestPending: () => ({
    type: MESSAGES_PENDING
  }),

  /**
   * requestSuccess action creator
   *
   * @param {string} locale - locale identifier to request
   * @param {array} response
   * @property {object} responseItem
   * @property {string} responseItem.module
   * @property {string} responseItem.page
   * @property {object} responseItem.messages
   */
  requestSuccess: (locale, response) => ({
    type: MESSAGES_SUCCESS,
    locale,
    response
  }),

  /**
   * requestError action creator
   *
   * @param {object} error - new Error('reason')
   */
  requestError: (error) => ({
    type: MESSAGES_ERROR,
    error
  })
});

export default LanguageActionsFactory;
