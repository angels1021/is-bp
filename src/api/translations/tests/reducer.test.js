/**
 * Test reducer
 */
import translationsReducer from '../reducer';
import { requestMessages, messageSuccess, messagesError } from '../actions';

describe('API translations reducer', () => {
  let state;
  beforeEach(() => {
    state = translationsReducer(undefined, {});
  });

  it('should return the initial state', () => {
    expect(state).toMatchSnapshot();
  });

  describe('MESSAGES_REQUEST', () => {
    it('should reset error and set pending to the request', () => {
      // assert
      const locale = 'en';
      const request = [
        {
          module: 'app',
          page: 'common',
          messages: {
            'app.common.text': 'text'
          }
        }
      ];
      const expectedResults = state
        .set('error', false)
        .set('pending', { locale, request });

      // act
      const result = translationsReducer(state, requestMessages(locale, request));

      // assert
      expect(expectedResults).toEqual(result);
    });
  });

  describe('MESSAGES_SUCCESS', () => {
    it('should reset error,reset pending and update messages', () => {
      // assert
      const locale = 'he';
      const response = [
        {
          module: 'app',
          page: 'common',
          messages: {
            'app.common.text': 'text'
          }
        }
      ];

      const expectedResults = state
        .set('error', false)
        .set('pending', false)
        .mergeIn(['messages', locale, 'app', 'common'], response[0].messages);

      // act
      const result = translationsReducer(state, messageSuccess(locale, response));
      // assert
      expect(expectedResults).toEqual(result);
    });
  });

  describe('MESSAGES_FAIL', () => {
    it('should reset pending and set error with the error received', () => {
      // assert
      const error = new Error('some error');

      const expectedResults = state
        .set('error', error)
        .set('pending', false);

      // act
      const result = translationsReducer(state, messagesError(error));

      // assert
      expect(expectedResults).toEqual(result);
    });
  });
});
