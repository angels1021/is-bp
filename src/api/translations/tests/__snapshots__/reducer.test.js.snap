// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`API translations reducer should return the initial state 1`] = `
Object {
  "error": false,
  "messages": Object {
    "en": Object {
      "app": Object {
        "common": Object {},
      },
    },
  },
  "pending": false,
}
`;
